driver/src/main.rs

```diff
@@ -0,0 +1,195 @@
+extern crate models;
+extern crate mongodb;
+extern crate tiny_keccak;
+extern crate byteorder;
+extern crate rustc_hex;
+extern crate web3;
+
+use web3::futures::Future;
+use web3::contract::{Contract, Options};
+use web3::types::{Address, H256, U256};
+
+use std::fs;
+use std::io;
+use mongodb::bson;
+use mongodb::{Client, ThreadedClient};
+use mongodb::db::ThreadedDatabase;
+use std::time::Duration;
+use std::thread;
+
+
+fn get_current_balances(client: Client) -> Result<models::State, io::Error>{
+
+    let coll = client.db("dfusion").collection("CurrentState");
+
+    // Find the document and receive a cursor
+    let cursor = coll.find(None, None)
+        .ok().expect("Failed to execute find.");
+    
+    let docs: Vec<_> = cursor.map(|doc| doc.unwrap()).collect();
+
+    if docs.len() == 0 {
+        println!("Error: No CurrentState in the dfusion database");
+    }
+
+    if docs.len() != 1 {
+        println!("Error: There should be only one CurrentState");
+    }
+
+    let json: serde_json::Value = serde_json::to_value(&docs[0]).expect("Failed to parse json");;
+    let mut query=String::from(r#" { "curState": "#);
```

Comment:
It seems dangerous to fetch information by string comparison. Is there no way to get by key?

Reply:
How else would you do it? I can not imagine a better solution. What would be the key value? The object id?

Reply:
I can look into rust's json comprehension, but in python it would look like;

```python
import json
with open('some.json') as f:
    my_info = json.load(f)

curr_state = my_info['curState']
```

Reply:
According to this, one would use `json.get("curState")`

https://stackoverflow.com/questions/30292752/how-do-i-parse-a-json-file

Reply:
```rust
let file = fs::File::open("text.json")
    .expect("file should open read only");
let json: serde_json::Value = serde_json::from_reader(file)
    .expect("file should be proper JSON");
let first_name = json.get("FirstName")
    .expect("file should have FirstName key");
```

Reply:
But I am doing something different here. I am trying to find something specific from the data base and not catch all data from json and then later get some part of the data

Reply:
ok that sounds alright then..